### PR TITLE
Add bullet, sankey and waterfall chart

### PR DESCRIPTION
**Pull request by suro1995**
1. bullet
image

2. sankey
image

3. waterfall
image

### CODE REVIEW

File: ui/.eslintrc [hunk 1]
```diff
@@ -33,6 +33,7 @@
     ],
     "import/prefer-default-export": 0,
     "linebreak-style": 0,
-    "jsx-a11y/anchor-is-valid": 0
+    "jsx-a11y/anchor-is-valid": 0,
+    "no-mixed-operators": 0
```
Comment (lizho-ling) [inline ui/.eslintrc]: This is a rule to help us make good use of operators. pls do not disable it, instead, you should follow the rule.

File: ui/src/component/waterfall.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      yAxis: [
+        {
+          type: 'value',
+        },
+      ],
+      series: (() => {
+        const assistant = [];
+        const positive = [];
+        const negtive = [];
+        //               = assistant[i-1]        (num >= 0)
```
Comment (lizho-ling) [inline ui/src/component/waterfall.js]: The description doesn't seem correct to me.

File: ui/stories/sankey.stories.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { storiesOf } from '../.storybook/facade';
+import Sankey from '../src/component/sankey';
+
+const source = [
+  ['key', 'a', 'b', 'c', 'd', 'e', 'f'],
```
Comment (lizho-ling) [inline ui/stories/sankey.stories.js]: To me , below data structure makes more sense, it demonstrates useful meaning of the data. With below data structure, the data processing in chart component will also be easy.

```
[Source, Target, Value]
['a', 'b', 2]
['c', 'd', e'']
...
```

File: ui/src/component/sankey.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            for (let j = 1; j < source.length; j += 1) {
+              if (source[i][j] > 0) {
+                links.push({
+                  source: source[0][i],
+                  target: source[0][j],
+                  value: source[i][j],
+                });
+              }
+            }
+          }
```
Comment (lizho-ling) [inline ui/src/component/sankey.js]: Pls see my comment in sankey.stories. js

File: ui/src/component/bullet.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            symbolOffset: ['-30%', 0],
+            z: 20,
+          },
+          default: (idx, tot) => ({
+            type: 'bar',
+            stack: 'total',
+            barWidth: 30,
+            color: (rgb => `rgb(${rgb}, ${rgb}, ${rgb})`)(_.round((idx - 2) / (tot - 2) * (255 - 153) + 153)),
+          }),
+        };
```
Comment (lizho-ling) [inline ui/src/component/bullet.js]: More the map out of _.map,

File: ui/src/component/bullet.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            },
+            symbolSize: [20, 3],
+            symbolOffset: ['-30%', 0],
+            z: 20,
+          },
+          default: (idx, tot) => ({
+            type: 'bar',
+            stack: 'total',
+            barWidth: 30,
+            color: (rgb => `rgb(${rgb}, ${rgb}, ${rgb})`)(_.round((idx - 2) / (tot - 2) * (255 - 153) + 153)),
```
Comment (lizho-ling) [inline ui/src/component/bullet.js]: If you define color in this  way, we will have no way to do theme for this chart.

Comment (suro1995) [reply ui/src/component/bullet.js]: This is not to hard code color here, but use a formula to calculate the gradient color.

File: ui/src/component/bullet.js (hunk 5 above)
Comment (suro1995) [reply ui/src/component/bullet.js]: Fixed

File: ui/stories/sankey.stories.js (hunk 3 above)
Comment (suro1995) [reply ui/stories/sankey.stories.js]: Think about the data format we defined before:
```js
export const source = [
  ['product', '2015', '2016', '2017'],
  ['Matcha Latte', 43.3, 85.8, 93.7],
  ['Milk Tea', 83.1, 73.4, 55.1],
  ['Cheese Cocoa', 86.4, 65.2, 82.5],
  ['Walnut Brownie', 72.4, 53.9, 39.1],
];
```
From this definition we have:
1. The first row indicates dimensions(y axis) 
2. The first column indicates categories(x axis).
3. The values are defined as a 2-dim matrix.

So in order to uniform the data format, it is better to follow this data structure and do not create a special format for this type of chart.

File: ui/src/component/waterfall.js (hunk 2 above)
Comment (suro1995) [reply ui/src/component/waterfall.js]: Fixed

File: ui/.eslintrc (hunk 1 above)
Comment (suro1995) [reply ui/.eslintrc]: Use eslint-disable-line instead of disable it globally.

**Pull request merged**